manager/allocator/allocator.go

```diff
@@ -27,6 +28,8 @@ type Allocator struct {
 	stopChan chan struct{}
 	// doneChan is closed when the allocator is finished running.
 	doneChan chan struct{}
+
+	ingressNetwork *api.Network
```

Comment:
I have just started to look at this code as well, not very familiar yet.
I agree it should not be a package variable, if in future we envision an ingress network per tenant.

But at first glance I am thinking `ingressNetwork` should be part of `netCtx`. 
Seems like all the network allocation operational information is to be held in `netCtx`, while the other fields at this level are infrastructure fields (store, ballot, stopChan,...)

WDYT, does it make sense ?